### PR TITLE
Minor Bug Fixes 1 for 1.0.0

### PR DESCRIPTION
**Pull request by webadict**
-Fixed various secrets from not working correctly.
-Fixed a few Deck-building issues. Decks should now allow adding up to 60 cards (but finishing a deck with more than 30 cards removes some at random.)
-Fixed Gorehowl to no longer proc on opponent attacks.
-Added Mega-Windfury visual.
-Fixed Blood Knight from gaining far too many stats.
-Fixed summons to summon in the right places for the most part.

### CODE REVIEW

**Comment by webadict** [comment]
I'm probably gonna close this PR for a bit... I've got some strange but good changes I'm gonna make and test before I add them, including some big Bolf Ramshield fixes (Which is clearly the important thing on everyone's mind.)

However, this should add the incredibly powerful PreDamageEvent we've been missing. Things that should happen in this event are Ice Block, Bolf Ramshield, Animated Armor, and... *sigh* Cursed Blade... And getting these all to work together would be great!

Also, Cursed Blade is probably the most annoying interaction I've ever seen in a card. Just putting that one out there. And here's why:

"Cursed Blade's effect is to double the reported damage of Damage Events targeting your hero. This leads to the following interactions:
    The final damage you take will be doubled, after all other modifiers (like Spell Damage, Prophet Velen and Fallen Hero) are taken into account. This also means that Mistress of Pain attacking an enemy hero with Cursed Blade will heal for the final (doubled) damage.[2]
    If you have Cursed Blade and Ice Block, Ice Block will take into account the double damage from Cursed Blade when deciding to trigger or not.
    If you have Cursed Blade and Bolf Ramshield, Bolf Ramshield takes double damage when intercepting, regardless of play order.[3]
    If you have Cursed Blade and Animated Armor, Animated Armor will set the Damage Event's damage to 1, but Cursed Blade's effect will double its damage whenever it is read, so you will still take 2 damage (Bolf will still intercept 2 damage, etc). (In addition, normally Animated Armor does not trigger if you would take 1 damage, but Cursed Blade's effect doubles the reported damage to 2, so Animated Armor will trigger for no effect in this scenario.)[4]"

Basically, the interaction happens before and after every single other Pre-Damage Event, which makes very little sense to me, and is difficult to follow the same rules as the other Events. But, at least this fixes up some of the issues we've been having with adding some newer cards.

**Comment by webadict** [comment]
Alright. I've added the Pre-Damage Event, so now events properly activate (and damage modifiers can now be added properly!)

**Comment by webadict** [comment]
Oh, that's for Bolf Ramshield... I didn't want to modify too many things, but Bolf is actually supposed to injure himself! So, the only thing I could think of is what I implemented. It's... mostly just so that it works. It could be fixed. I was just doing a quick change, feel free to make that work better.

**Comment by mayuso** [comment]
You can actually add the text `Fixed #XX` to your commit to close an issue you fixed. This will close the issue and also help us identify what Issues need attention and what Issues you already solved.

(XX being the number of the Issue).

**Comment by webadict** [comment]
Thanks, I'll try to remember that for next time.

**Comment by mayuso** [comment]
No, thanks to you for your awesome job :)
